Sources_v3/StreamChatUI/Utils/ChannelNamer.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+
+open class ChannelNamer {
+    public required init() {}
+    
+    open func name<ExtraData: ExtraDataTypes>(for channel: _ChatChannel<ExtraData>, as currentUserId: UserId?) -> String {
+        if let channelName = channel.name, !channelName.isEmpty {
+            // If there's an assigned name and it's not empty, we use it
+            return channelName
+        } else if channel.cid.id.hasPrefix("!members") {
```

Comment:
There's `isDirectMessageChannel` flag on channel but it checks `member.count == 2`, this check is not required by backend, if we only pass user id's and not cid during channel creation, backend will generate a cid for us, and we assume this channel is "direct message", regardless of participant count

Reply:
@b-onc I think we should fix `isDirectMessageChannel` to check only for the prefix. The `member.count == 2` is not necessary imo. Clients can easily do it if they need to.